Automattic/jetpack
# Gutenberg: Try Tiled Gallery block

⚠️ _DO NOT MERGE!_

---

This branch is for testing out Tiled Gallery block in Gutenberg (PR https://github.com/Automattic/wp-calypso/pull/27159).

<img width="690" alt="screen shot 2018-09-12 at 17 34 10" src="https://user-images.githubusercontent.com/87168/45433729-7569bd80-b6b5-11e8-8ad1-810b82b3b471.png">

![image](https://user-images.githubusercontent.com/87168/45558764-59972080-b849-11e8-9da8-04c3c1fcd3ea.png)

#### Features
- Loads images via Photon CDN: currently this happens in editor mode, but since it's kinda slow it could happen only at the theme-side. Uses Photon's "fit" feature (https://developer.wordpress.com/docs/photon/).
- Supports sidebar settings  for:
  - "link to" (where each image is linked to)
  - "columns"
- Supports captions
- Supports adding/removing/re-organising images via media-gallery:
    ![image](https://user-images.githubusercontent.com/87168/45559525-5f8e0100-b84b-11e8-9e0a-1c376bbf8972.png)
- Supports transforming the block to/from core gallery

#### Notes

- Showing the gallery at the theme side is still broken in this version.

- No "slides" or "circle" styles. Slideshow we should probably add as a separate block but circles is doable :thumbsup:

- Adding/removing images is a bit wonky experience when using "upload" button. "Media gallery" button seems to work fine. 

- Does not support re-ordering by dragging images

- Captions are not pretty — core gallery has much nicer ones these days

- What features from the core gallery we want to have? In core gallery you can edit images + captions directly in the block. In tiled gallery you need to hit "edit" and edit in different interface.

- Icon is pretty similar with core-gallery (tiled gallery on right)
    ![image](https://user-images.githubusercontent.com/87168/45559098-3c168680-b84a-11e8-829a-b8428ddb50c0.png)


## Testing

- Open:
    ```
    https://jurassic.ninja/create/?jetpack-beta&branch=try/block-tiled-gallery&shortlived&wp-debug-log&gutenberg
    ```
- Create a new post
- Add tiled gallery block
- Test both "upload" and "media gallery" buttons, latter works a bit better.
- Make sure to open sidebar to see extra options
